### ISSUE TITLE
rngd: Fails to start with "can't open any entropy source"

### ISSUE THREAD

**Issue by fuzzy-id** [NixOS/nixpkgs#31648]
## Issue description

`rngd` does not start on my system since the latest update.
```
● rngd.service - Hardware RNG Entropy Gatherer Daemon
   Loaded: loaded (/nix/store/wad7xz5kk8kc1c0sgc8852pr8821ah68-unit-rngd.service/rngd.service; linked; vendor preset: enabled)
   Active: failed (Result: exit-code) since Tue 2017-11-14 06:51:52 CET; 14min ago
  Process: 740 ExecStart=/nix/store/74xsgizq0d7x6c0cl0p7y34ngwihxiji-rng-tools-5/sbin/rngd -f -v (code=exited, status=1/FAILURE)
 Main PID: 740 (code=exited, status=1/FAILURE)

Nov 14 06:51:51 bahru systemd[1]: Started Hardware RNG Entropy Gatherer Daemon.
Nov 14 06:51:51 bahru rngd[740]: Unable to open file: /dev/tpm0
Nov 14 06:51:51 bahru rngd[740]: can't open any entropy source
Nov 14 06:51:51 bahru rngd[740]: Maybe RNG device modules are not loaded
Nov 14 06:51:52 bahru systemd[1]: rngd.service: Main process exited, code=exited, status=1/FAILURE
Nov 14 06:51:52 bahru systemd[1]: rngd.service: Unit entered failed state.
Nov 14 06:51:52 bahru systemd[1]: rngd.service: Failed with result 'exit-code'.
```
### Steps to reproduce

Nothing special, just reboot.

## Technical details

* System: `17.09.2055.c19136c4c9`
* Nix version: `1.11.15`
* Nixpkgs version: `17.09.2055.c19136c4c9`
* Sandboxing enabled: yes

**Comment by MrSorcus** [comment]
Confirm.

**Comment by astrojhgu** [comment]
Confirmed, this problem still exists in version 18.09

**Comment by c0bw3b** [comment]
If `rngd` fails to start entirely then you are in context where _no_ **hardware** source of entropy is available.
You may disable it (`security.rngd.enable = false`) and use `haveged` instead (`services.haveged.enable = true`)

If `rngd` just logs _Unable to open file: /dev/tpm0_ at startup but remains active then you just don't have a TPM module in your system and rngd is using another hardware entropy source such as DRNG (aka Intel RDRAND). I can reproduce this situation in a VBox VM for example.

More on this topic:
https://www.certdepot.net/rhel7-get-started-random-number-generator/

**Comment by devhell** [comment]
Can confirm that the service is failing to start and this is on bare-metal:
```
Available entropy sources:
	Intel/AMD hardware rng
	DRNG
```

**Comment by c0bw3b** [comment]
If you have rng-tools v6.6 but your service unit is still trying to run it with `rngd -f -v` then it will be solved once #49778 reaches your channel.

**Comment by step21** [comment]
For vms, this should be added to the virtualbox config instructions. I mean, there is a special section for virtualbox, but this is not in it. Or is it somehow possible to start anyway? My vbox install just hangs on starting rngd.
